private/pkg/protodescriptor/protodescriptor.go

```diff
@@ -161,6 +166,28 @@ func ValidateCodeGeneratorRequestExceptFileDescriptorProtos(request *pluginpb.Co
 	if err := ValidateProtoPaths("CodeGeneratorRequest.FileToGenerate", request.FileToGenerate); err != nil {
 		return err
 	}
+	if len(request.SourceFileDescriptors) != len(request.FileToGenerate) {
```

Comment:
We can't be guaranteed in this package that the input will be generated by a recent compiler, it is valid for the length of `source_file_descriptors` to be 0 here.

Re: PR comment, I am likely just going to have all this validation live in `protoplugin` and import it here, so we don't need to redo it anyways (although I will add the duplicate check over there).

Reply:
Ah, right. My bad. I wasn't entirely sure where this was used. But it makes sense that it could be used in the recipient of a request, not just the producer, which means it must be able to handle an older form of the request.

Reply:
The issue hasn't been fixed, we'll still fail for old payloads.

Reply:
> The issue hasn't been fixed, we'll still fail for old payloads.

Because I hadn't pushed any updates 😛 

Fixed now. I just removed most of the extra validation. I realized that the simple check that was here wasn't right anyway because `source_file_descriptors` is it not _specified_ to be in the same order as `file_to_generate`. And it's also not in the same order in practice, neither from the code I just wrote for `buf generate` nor for `protoc`. Both (`protoc` and `buf`) actually build it in the same order as `proto_file`, but pruned down to only the paths in `file_to_generate`.

So doing the _right_ check is a little more complicated and seemed like it warranted a separate PR with a bevy of new tests to make sure it's right. So I can push a PR up for that later.